### PR TITLE
UI: Fix the order of addons appearing in prebuilt manager

### PR DESCRIPTION
**Pull request by ndelangen**
The addons did not appear in the same order as before, meaning the order in which they are defined in the user's `main.ts`.

I discovered the reason why:
When I have esbuild write to the file-system, and then read the file-system... The order in which I get them when I read isn't the same as when I wrote.

I changed the behavior to have esbuild NOT write the files to the file-system, when doing so the return value of esbuild contains a `outputFiles` property. (it does not contain this property when you configure it to write files).

This array of files is in the proper order and contains the correct path.

### CODE REVIEW

File: code/lib/builder-manager/src/index.ts [hunk 1]
```diff
@@ -118,15 +118,24 @@ const starter: StarterFunction = async function* starterGeneratorFn({
   router.use(`/sb-addons`, express.static(addonsDir));
   router.use(`/sb-manager`, express.static(coreDirOrigin));
 
-  const addonFiles = readDeep(addonsDir);
+  const files = await Promise.all(
```
Comment (shilman) [inline code/lib/builder-manager/src/index.ts:121]: Refactor this into a function

**Review by shilman — CHANGES_REQUESTED** [review]
Small request

File: code/lib/builder-manager/src/index.ts (hunk 1 above)
Comment (ndelangen) [reply code/lib/builder-manager/src/index.ts]: done!

File: code/lib/builder-manager/src/index.ts [hunk 2]
```diff
@@ -19,11 +19,12 @@ import {
   ManagerBuilder,
   StarterFunction,
 } from './types';
-import { readDeep } from './utils/directory';
 import { getData } from './utils/data';
 import { safeResolve } from './utils/safeResolve';
+import { readOrdererFiles } from './utils/files';
```
Comment (shilman) [inline code/lib/builder-manager/src/index.ts:24]:
```suggestion
import { readOrderedFiles } from './utils/files';
```

File: code/lib/builder-manager/src/index.ts [hunk 3]
```diff
@@ -118,15 +120,16 @@ const starter: StarterFunction = async function* starterGeneratorFn({
   router.use(`/sb-addons`, express.static(addonsDir));
   router.use(`/sb-manager`, express.static(coreDirOrigin));
 
-  const addonFiles = readDeep(addonsDir);
+  const { cssFiles, jsFiles } = await readOrdererFiles(addonsDir);
```
Comment (shilman) [inline code/lib/builder-manager/src/index.ts:123]:
```suggestion
  const { cssFiles, jsFiles } = await readOrderedFiles(addonsDir);
```

File: code/lib/builder-manager/src/index.ts [hunk 4]
```diff
@@ -182,22 +185,23 @@ const builder: BuilderFunction = async function* builderGeneratorFn({ startTime,
   yield;
 
   const managerFiles = copy(coreDirOrigin, coreDirTarget);
-  const addonFiles = readDeep(addonsDir);
+  const { cssFiles, jsFiles } = await readOrdererFiles(addonsDir);
```
Comment (shilman) [inline code/lib/builder-manager/src/index.ts:188]:
```suggestion
  const { cssFiles, jsFiles } = await readOrderedFiles(addonsDir);
```

File: code/lib/builder-manager/src/utils/files.ts [hunk 5]
```diff
@@ -0,0 +1,15 @@
+import { writeFile, ensureFile } from 'fs-extra';
+import { compilation } from '../index';
+
+export async function readOrdererFiles(addonsDir: string) {
```
Comment (shilman) [inline code/lib/builder-manager/src/utils/files.ts:4]:
```suggestion
export async function readOrderedFiles(addonsDir: string) {
```

**Review by shilman — APPROVED** [review]
Fix typo

**Pull request merged**